src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/persistence/PersistenceMapKeyDiagnosticsCollector.java

```diff
@@ -37,81 +37,56 @@ protected String getDiagnosticSource() {
-                                hasMapKeyAnnotation = true;
-                            else if (PersistenceConstants.MAPKEYCLASS.equals(matchedAnnotation))
-                                hasMapKeyClassAnnotation = true;
-                            else if (PersistenceConstants.MAPKEYJOINCOLUMN.equals(matchedAnnotation)) {
-                                mapKeyJoinCols.add(annotation);
-                            }
-                        }
-                    }
-                    if (hasMapKeyAnnotation && hasMapKeyClassAnnotation) {
-                        // A single field cannot have the same
-                        diagnostics.add(createDiagnostic(field, unit,
-                                Messages.getMessage("MapKeyAnnotationsNotOnSameField"),
-                                PersistenceConstants.DIAGNOSTIC_CODE_INVALID_ANNOTATION, null,
-                                DiagnosticSeverity.Error));
-                    }
-                    if (mapKeyJoinCols.size() > 1) {
-                        validateMapKeyJoinColumnAnnotations(mapKeyJoinCols, field, unit, diagnostics);
-                    }
+                    collectDiagnostics(unit, diagnostics, type, field);
+                }
+            }
+        }
+    }
+
+    private void collectDiagnostics(PsiJavaFile unit, List<Diagnostic> diagnostics,
+                                    PsiClass type, PsiJvmModifiersOwner fieldOrProperty) {
+        List<PsiAnnotation> mapKeyJoinCols = new ArrayList<PsiAnnotation>();
+        boolean hasMapKeyAnnotation = false;
+        boolean hasMapKeyClassAnnotation = false;
+        PsiAnnotation[] allAnnotations = fieldOrProperty.getAnnotations();
+        for (PsiAnnotation annotation : allAnnotations) {
+            String matchedAnnotation = getMatchedJavaElementName(type, annotation.getQualifiedName(),
+                    PersistenceConstants.SET_OF_PERSISTENCE_ANNOTATIONS);
+            if (matchedAnnotation != null) {
+                if (PersistenceConstants.MAPKEY.equals(matchedAnnotation))
+                    hasMapKeyAnnotation = true;
+                else if (PersistenceConstants.MAPKEYCLASS.equals(matchedAnnotation))
+                    hasMapKeyClassAnnotation = true;
+                else if (PersistenceConstants.MAPKEYJOINCOLUMN.equals(matchedAnnotation)) {
+                    mapKeyJoinCols.add(annotation);
                 }
             }
         }
+        if (hasMapKeyAnnotation && hasMapKeyClassAnnotation) {
+            // A single field or property cannot have the same
+            diagnostics.add(createDiagnostic(fieldOrProperty, unit,
+                    Messages.getMessage("MapKeyAnnotationsNotOnSameField"),
+                    PersistenceConstants.DIAGNOSTIC_CODE_INVALID_ANNOTATION, null,
+                    DiagnosticSeverity.Error));
+        }
+        // If we have multiple MapKeyJoinColumn annotations on a single method or property we must
```

Comment:
Just noting that the `fieldOrProperty` param name seems off based on this comment.

Reply:
It's the correct terminology. See the Jarkata Persistence specification.

Reply:
I did notice typo. it should say "... on a single field or property ..." instead of "... on a single method or property ...". I'll fix that.